tests/test_scrape_mensa.py

```diff
@@ -15,3 +50,56 @@ def test_read_menu_page():
 def test_failing_read_menu_page():
     with pytest.raises(requests.exceptions.HTTPError, match="500 Server Error"):
         read_page('http://menu.mensen.at/WRONG_URI')
+
+
+def test_parse_menu_basic(day):
+    items = [
+        ("Spam", "10,11"),
+        ("Eggs and Spam", "3,40"),
+    ]
+    page = generate_test_html([day], items)
+    assert (
+        parse_menu(page, day)
+        ==
+        [f"{food} € {price}" for food, price in items]
+    )
+
+
+def test_parse_menu_finds_date(day):
+    day_index = 7
+    days = [day + datetime.timedelta(i) for i in range(10)]
+    page = generate_test_html(
+        days=days,
+        items=[("Spam", "10,11")],
+        day_index=day_index,
+    )
+
+    menu_items = parse_menu(page, days[day_index])
+
+    assert len(menu_items) == 1
+
+
+def test_parse_menu_ignores_hints(day):
+    page = generate_test_html([day], items=[
+        ("Herzlich Willkommen", ""),
+        ("Spam", "10,11"),
+        ("Tages - Empfehlung", ""),
+    ])
+
+    menu_items = parse_menu(page, day)
+
+    assert len(menu_items) == 1
+
+
+def test_parse_highlights_vegetarian(day):
+    vegy_type = next(iter(VEGY_TYPES))
```

Comment:
Did you plan to run the test for all `VEGY_TYPES` or just for the first one? I guess both is fine, but looks a bit like it was intended to use a loop or a fixture to me.

Reply:
I think there are different approaches, i tend to test for examples if the pattern is the same for all values of the datatype.

Here we could test for all possible values, but I'm not sure if there's really a benefit in that.

Reply:
Ok, I understand. Thanks for the explaination!